modules/cnf-programming-the-n3000-operator.adoc

```diff
@@ -0,0 +1,351 @@
+{"level":"info","ts":1608054338.9004142,"logger":"daemon.fpgaManager.ProgramFPGA","msg":"Starting","pci":"0000:1b:00.0"}
+{"level":"info","ts":1608056309.9367146,"logger":"daemon.fpgaManager.ProgramFPGA","msg":"Program FPGA completed, start new power cycle N3000 ...","pci":"0000:1b:00.0"}
+{"level":"info","ts":1608056333.3528838,"logger":"daemon.drainhelper.Run()","msg":"worker function - end","performUncordon":true}
+...
+----
++
+The log file indicates the following flow of events:
+
+* The bitstream is downloaded and validated.
+* The node is drained and no workload is able to run during this time.
+* Flashing is started:
+** The bitstream is flashed into the card.
+** The bitstream is applied.
+* After flashing is complete the PCI device or devices on the node or nodes are reloaded. The Intel wireless Operator is now able to find the new flashed device or devices.
+
+. Verify the status after the FPGA user bitstream update is complete:
++
+[source,terminal]
+----
+oc get n3000node
+----
++
+.Example output
+[source,terminal]
+----
+NAME             FLASH
+node1            Succeeded
+----
+
+. Verify that the bitstream ID of the card has changed:
++
+[source,terminal]
+----
+oc get n3000node node1 -o yaml
+----
++
+.Example output
+[source,yaml]
+----
+status:
+      conditions:
+          - lastTransitionTime: "2020-12-15T18:18:53Z"
+            message: Flashed successfully <1>
+            observedGeneration: 2
+            reason: Succeeded
+            status: "True"
+            type: Flashed
+      fortville:
+      - N3000PCI: 0000:1b:00.0
+        NICs:
+        - MAC: 64:4c:36:11:1b:a8
+          NVMVersion: 7.00 0x800052b0 0.0.0
+          PCIAddr: 0000:1a:00.0
+          name: Ethernet Controller XXV710 Intel(R) FPGA Programmable Acceleration Card N3000 for Networking
+        - MAC: 64:4c:36:11:1b:a9
+          NVMVersion: 7.00 0x800052b0 0.0.0
+          PCIAddr: 0000:1a:00.1
+          name: Ethernet Controller XXV710 Intel(R) FPGA Programmable Acceleration Card N3000 for Networking
+        - MAC: 64:4c:36:11:1b:ac
+          NVMVersion: 7.00 0x800052b0 0.0.0
+          PCIAddr: 0000:1c:00.0
+          name: Ethernet Controller XXV710 Intel(R) FPGA Programmable Acceleration Card N3000 for Networking
+        - MAC: 64:4c:36:11:1b:ad
+          NVMVersion: 7.00 0x800052b0 0.0.0
+          PCIAddr: 0000:1c:00.1
+          name: Ethernet Controller XXV710 Intel(R) FPGA Programmable Acceleration Card N3000 for Networking
+      fpga:
+      - PCIAddr: 0000:1b:00.0 <2>
+        bitstreamId: "0x2315842A010601" <3>
+        bitstreamVersion: 0.2.3
+        deviceId: "0x0b30 <4>
+----
++
+<1> The message field indicates the device is successfully flashed.
+<2> Use this field to check that the correct PCI address of the card is programmed.
+<3> Use this field to verify that the `bitstreamId` is changed. Check this against the output obtained from step 5.
+<4> The current deviceID of the bitstream inside the card exposed to the system.
```

Comment:
<1> The `message` field...
<2> The `PCIAddr` field indicates the PCI address of the card.
<3> The `bitstreamId` field indicates the updated bitstream ID.
<4> The `deviceId` field indicates that device ID of the bitstream...